pycbc/strain.py

```diff
@@ -70,14 +117,25 @@ def detect_loud_glitches(strain, psd_duration=4., psd_stride=2.,
     strain_pad[pad_start:pad_end] = strain[:]
 
     logging.info('Autogating: estimating PSD')
-    psd = pycbc.psd.welch(strain, seg_len=int(psd_duration*strain.sample_rate),
-                          seg_stride=int(psd_stride*strain.sample_rate),
+
+    # get the longest portion of strain over which the PSD can be estimated
+    psd_seg_length = int(psd_duration * strain.sample_rate)
+    psd_stride_length = int(psd_stride * strain.sample_rate)
+    psd_estim_length = psd_seg_length
+    while psd_estim_length < (len(strain) - 2 * corrupt_length - psd_stride_length):
+        psd_estim_length += psd_stride_length
+    psd_estim_start = len(strain) / 2 - psd_estim_length / 2
+    psd_estim_end = psd_estim_start + psd_estim_length
```

Comment:
I think this duplicates code in the PSD module. The PSD module no longer requires data length to be a multiple of seg_len.